api_app/serializers.py

```diff
@@ -77,18 +80,194 @@ class _AbstractJobCreateSerializer(rfs.ModelSerializer):
+            try:
+                config = analyzer_dataclasses.get(a_name, None)
+
+                if not config:
+                    if not run_all:
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't run: not available in configuration"
+                        )
+                    # don't add warning if run_all
+                    continue
+
+                if not config.is_ready_to_use:
+                    raise NotRunnableAnalyzer(
+                        f"{a_name} won't run: is disabled or unconfigured"
+                    )
+
+                if serialized_data["is_sample"]:
+                    if not config.is_type_file:
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because does not support files."
+                        )
+                    if not config.is_filetype_supported(
+                        serialized_data["file_mimetype"]
+                    ):
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because mimetype "
+                            f"{serialized_data['file_mimetype']} is not supported."
+                        )
+                else:
+                    if not config.is_type_observable:
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because "
+                            f"it does not support observable."
+                        )
+
+                    if not config.is_observable_type_supported(
+                        serialized_data["observable_classification"]
+                    ):
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because "
+                            f"it does not support observable type "
+                            f"{serialized_data['observable_classification']}."
+                        )
+
+                if tlp != TLP.WHITE and config.leaks_info:
+                    raise NotRunnableAnalyzer(
+                        f"{a_name} won't be run because it leaks info externally."
+                    )
+                if tlp == TLP.RED and config.external_service:
+                    raise NotRunnableAnalyzer(
+                        f"{a_name} won't be run because you"
+                        f" filtered external analyzers."
+                    )
+            except NotRunnableAnalyzer as e:
+                if run_all:
+                    # in this case, they are not warnings but
+                    # expected and wanted behavior
+                    logger.debug(e)
+                else:
+                    logger.warning(e)
+                    warnings.append(str(e))
+            else:
+                cleaned_analyzer_list.append(a_name)
+
+        return cleaned_analyzer_list
+
+    @staticmethod
+    def filter_connectors(serialized_data: Dict, warnings: List[str]) -> List[str]:
```

Comment:
Since you have decided to move this stuff, i would love that we remove the fact that we modify the parameters of the function, inside the function itself.
What i mean is pretty much change the signature to 
```python3
def filter_analyzers(serialized_data: Dict) -> Tuple[List[str], List[str]]:
```
And concatenate the `warnings` in the caller